Doc/library/functools.rst

```diff
@@ -638,6 +638,23 @@ The :mod:`functools` module defines the following functions:
    on the wrapper function). :exc:`AttributeError` is still raised if the
    wrapper function itself is missing any attributes named in *updated*.
 
+   When assigning the ``__annotations__`` attribute (when it's part of the
+   ``assigned`` list parameter), it will be populated only if missing some
+   annotations as follows:
+
+   * If the wrapper function defines it's own return type and parameter
+     annotations, those will be preserved.
+
+   * If the wrapper function does not annotate the return type, it will be
+     copied from the wrapped if any.
+
+   * If the wrapped function does not annotate any parameters, the annotations
+     for them will be copied over from the wrapped if any.
+
+   That ensures that any custom annotations from the wrapper will not be
+   overridden by the wrapped, allowing to change the annotated types of
+   the wrapped function.
+
    .. versionadded:: 3.2
```

Comment:
Clearly you need a `versionadded:: 3.12` block for the new treatment of `__annotations__`.